internal/app/server/middleware/gzip.go

```diff
@@ -0,0 +1,90 @@
+package middleware
+
+import (
+	"compress/gzip"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+)
+
+type gzipWriter struct {
+	gin.ResponseWriter
+	writer *gzip.Writer
+}
+
+func (g *gzipWriter) Write(data []byte) (int, error) {
+	return g.writer.Write(data)
+}
+
+func (g *gzipWriter) WriteString(s string) (int, error) {
+	return g.writer.Write([]byte(s))
+}
+
+func shouldCompress(c *gin.Context) bool {
+	// Check if client accepts gzip encoding
+	if !strings.Contains(strings.ToLower(c.Request.Header.Get("Accept-Encoding")), "gzip") {
+		return false
+	}
+
+	// Get the content type, defaulting to the one that will be set
+	contentType := c.Writer.Header().Get("Content-Type")
+	if contentType == "" {
+		contentType = c.Request.Header.Get("Accept")
+	}
+
+	// Check if content type should be compressed
+	return strings.Contains(contentType, "application/json") ||
```

Comment:
[CHANGE REQUEST] Please think about this condition again.

Reply:
Done